app/controllers/rooms_controller.rb

```diff
@@ -25,11 +27,23 @@ def create
 
   def update
     respond_to do |format|
-      if room.update(room_params)
+      new_media = Media.create
+      new_media.upload.attach(room_params[:hero_image_upload])
+      room_params.delete("hero_image_upload")
+      room_params_for_update = {}.merge(
+        room_params,
+        {
+          hero_image: new_media
+        }
+      )
+
+      if room.update(room_params_for_update)
```

Comment:
@zspencer controller handling for the file upload and media creation before update. I'm not confident this approach is best practice for handling relationships in a single controller action. Or maybe there's a better way to setup the form and accepted parameters. But this helped me prove I could test upload files through the UI and create the records correctly.

Reply:
I'm cool with making it RESTier, and can work on this in the next iteration which can include prototyping an interface for managing multiple Room images with more concrete functional specs.